tests/mock_virtual_machine.h

```diff
@@ -19,24 +19,18 @@
 #define MULTIPASS_MOCK_VIRTUAL_MACHINE_H
 
 #include <gmock/gmock.h>
-#include <multipass/virtual_machine.h>
+
+#include "platform/backends/shared/base_virtual_machine.h"
 
 namespace multipass
 {
 namespace test
 {
-struct MockVirtualMachine : public multipass::VirtualMachine
+// This mocks a virtual machine, but not the methods of the base virtual machine.
+struct MockNonBaseVirtualMachine : public multipass::BaseVirtualMachine
```

Comment:
OK, now I understand what you were asking about. Your issue steams from the fact that you're really testing `BaseVirtualMachine`, not the daemon. These being unit tests, the daemon code itself could be perfectly tested with a mocked `get_all_ipv4`.

So, I think what you want is a `test_base_virtual_machine.h` to test `get_all_ipv4()` in. In that file (only useful there) you would have a special `StubBaseVirtualMachine : public multipass::BaseVirtualMachine`, with stubs for all the methods that need an implementation. Mocks are only useful if you want behavior to change (e.g. if you want `current_state` to return different things). Otherwise a stub is enough. `mock_virtual_machine.h` shouldn't have to change.

Reply:
Absolutely. I removed the tests from the daemon and created a new `BaseVMTest`. This way it was muuuuuch easier to test `get_all_ipv4()`.